InnerEye/ML/model_testing.py

```diff
@@ -409,72 +410,79 @@ def classification_model_test(config: ScalarModelBase,
     """
     posthoc_label_transform = config.get_posthoc_label_transform()
 
-    def test_epoch(checkpoint_paths: List[Path]) -> Optional[MetricsDict]:
-        pipeline = create_inference_pipeline(config=config,
-                                             checkpoint_paths=checkpoint_paths)
-
-        if pipeline is None:
-            return None
-
-        # for mypy
-        assert isinstance(pipeline, ScalarInferencePipelineBase)
-
-        ml_util.set_random_seed(config.get_effective_random_seed(), "Model Testing")
-        ds = config.get_torch_dataset_for_inference(data_split).as_data_loader(
-            shuffle=False,
-            batch_size=1,
-            num_dataload_workers=0
-        )
-
-        logging.info(f"Starting to evaluate model on {data_split.value} set.")
-        metrics_dict = create_metrics_dict_for_scalar_models(config)
-        for sample in ds:
-            result = pipeline.predict(sample)
-            model_output = result.posteriors
-            label = result.labels.to(device=model_output.device)
-            label = posthoc_label_transform(label)
-            sample_id = result.subject_ids[0]
-            compute_scalar_metrics(metrics_dict,
-                                   subject_ids=[sample_id],
-                                   model_output=model_output,
-                                   labels=label,
-                                   loss_type=config.loss_type)
-            logging.debug(f"Example {sample_id}: {metrics_dict.to_string()}")
-
-        average = metrics_dict.average(across_hues=False)
-        logging.info(average.to_string())
-
-        return metrics_dict
-
-    checkpoints_to_test = checkpoint_handler.get_checkpoints_to_test()
-
-    if not checkpoints_to_test:
+    checkpoint_paths = checkpoint_handler.get_checkpoints_to_test()
+    if not checkpoint_paths:
         raise ValueError("There were no checkpoints available for model testing.")
 
-    result = test_epoch(checkpoint_paths=checkpoints_to_test)
-    if result is None:
+    pipeline = create_inference_pipeline(config=config,
+                                         checkpoint_paths=checkpoint_paths)
+    if pipeline is None:
         raise ValueError("There was no single checkpoint file available for model testing.")
```

Comment:
Maybe update the exception message "inference pipeline couldn't be created" or sth like that

Reply:
Done!